### ISSUE TITLE
Handling pip with specified constraint file

### ISSUE THREAD

**Issue by matrixik** [hadolint/hadolint#229]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

In OpenStack we are using upper constraint file for specify max possible version of packaged to be installed.
Example: http://git.openstack.org/cgit/openstack/requirements/plain/upper-constraints.txt?h=master
The problem is that first I install some packages for waiting scripts and then install our app using using upper constraint file and if I install newer version than in upper constrain then this packages will be uninstalled and then installed in proper versions.

### Expected behavior

Avoid showing this error when upper constraint file is provided.

### Actual behavior
```
Dockerfile:54 DL3013 Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>`
```

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v1.9.0-0-g4c4881a
```

**Dockerfile** (if relevant)

```Dockerfile
FROM python:3.5.5-alpine3.7

RUN \
    pip3 install \
        pykafka \
        PyMySQL \
        Templer==1.1.4 \
        --constraint http://git.openstack.org/cgit/openstack/requirements/plain/upper-constraints.txt?h=master
```

Second example when installing Python app from source code:
```
RUN \
    pip3 install \
        /app/. \
        --requirement /app/requirements.txt \
        --constraint /app/upper-constraints.txt
```

**Additional environment details (OS, stack version, etc.)**
```
System:    Host: zybortd Kernel: 4.10.0-42-generic x86_64 (64 bit gcc: 6.3.0)
           Desktop: Gnome  (Gtk 3.22.11-0ubuntu3) Distro: Ubuntu 17.04
```

**Comment by lorenzo** [comment]
Thanks for the report, we'll look into adding support for that flag

**Comment by zemanlx** [comment]
@matrixik How can hadolint say that you specified your versions for packages? Is `--constraint` in `pip install` enough?
